### PR TITLE
Remove OIDC and SAML adapters for Wildfly/EAP ZIP downloads. Update documentation.

### LINKED ISSUE #24713: Remove and update some OIDC and SAML adapters for Wildfly/EAP
### Description

Wondering if for Keycloak 23 we should remove:

* WildFly OIDC adapter
* JBoss EAP 7 OIDC adapter
* JBoss EAP 7 SAML adapter

and only keep WildFly 29+ SAML adapter?

For 23 we could do a lightweight approach and only remove the uploads + the listing on the website. Then as a follow-up (so different issue than this one), we can delete the artifacts/etc when we don't need them for testsuite anymore.

### SAML Adapter updates

For Wildfly 29+ SAML adapter, our download and documentation will use Galleon instead of the installation of SAML adapter via ZIP. The `keycloak-saml-wildfly-adapter-dist` won't be provided in the distribution artifacts [1]. The keycloak download web page [2], documentation [3] and quickstart [4] will need to be updated to reflect this.

[1] https://github.com/keycloak/keycloak/releases
[2] https://www.keycloak.org/downloads
[3] https://www.keycloak.org/docs/latest/securing_apps/index.html#_saml-jboss-adapter-installation
[4] https://github.com/keycloak/keycloak-quickstarts/tree/main/jakarta/servlet-saml-service-provider

### OIDC Adapter updates

All those pages (release download artifacts, webpage download page, documentation) should be updated also for Keycloak OIDC adapters.

### Release notes

Should be updated for this (Also worth update for https://github.com/keycloak/keycloak/issues/12943, which is unrelated to this issue)

### Child issues

Keycloak web: https://github.com/keycloak/keycloak-web/issues/431
Keycloak quickstarts: https://github.com/keycloak/keycloak-quickstarts/issues/510

### PR DESCRIPTION
**Pull request by mposolda**
Closes #24713

### CODE REVIEW

File: docs/documentation/release_notes/topics/23_0_0.adoc [hunk 1]
```diff
@@ -1,3 +1,8 @@
+= OIDC and SAML adapter changes for WildFly and JBoss EAP
+
+OIDC adapter for the WildFly and JBoss EAP, which was deprecated in previous versions, is removed in this release. The SAML adapter ZIP download
+for WildFly/EAP is removed as it is being replaced by the Galleon feature pack. See the link:{adapterguide_link}[{adapterguide_name}] for the details.
```
Comment (stianst) [inline docs/documentation/release_notes/topics/23_0_0.adoc:4]: Perhaps add a note around Elytron OIDC?

File: docs/documentation/securing_apps/topics/oidc/java/jboss-adapter.adoc [hunk 2]
```diff
@@ -1,269 +1,16 @@
...
-        <bearer-only>true</bearer-only>
-    </secure-deployment>
-</subsystem>
-----
-
-===== Security domain
-
-The security context is propagated to the EJB tier automatically.
+For more details about how to integrate {project_name} with JakartaEE applications running on latest Wildfly/EAP, consider looking at the
+{quickstartRepo_link}[Keycloak Quickstart GitHub Repository] at the jakarta quickstarts.
```
Comment (stianst) [inline docs/documentation/securing_apps/topics/oidc/java/jboss-adapter.adoc:16]:
```suggestion
For more details on how to integrate {project_name} with JakartaEE applications running on latest Wildfly/EAP, take a look at the Jakarta EE quickstarts within the {quickstartRepo_link}[Keycloak Quickstart GitHub Repository].
```

File: docs/documentation/securing_apps/topics/saml/java/jboss-adapter/jboss_adapter_installation.adoc [hunk 3]
```diff
@@ -1,130 +1,16 @@
...
 
-ifeval::[{project_community}==true]
+{project_name} provided adapter ZIP download in the past, but it is not provided anymore. For the older WildFly versions, it is recommended to upgrade
+to newer WildFly/EAP and use Galleon. Otherwise, you will need to stick with the older {project_name} adapters, but those are not maintained and officially supported.
+For JBoss EAP 7, it is possible to stick with the Red Hat Single Sign-On 7.6 adapters, which is still supported.
 
-NOTE: We only test and maintain adapter with the most recent version of WildFly available upon the release. Once the new version of WildFly is released, the current adapters become deprecated and support for them will be removed after the next WildFly release.
-The other alternative is to switch your applications from WildFly to the JBoss EAP, as the JBoss EAP adapter is supported for a much longer period.
+For more details about how to integrate {project_name} with JakartaEE applications running on latest Wildfly/EAP, consider looking at the
+{quickstartRepo_link}[Keycloak Quickstart GitHub Repository] at the jakarta quickstarts.
```
Comment (stianst) [inline docs/documentation/securing_apps/topics/saml/java/jboss-adapter/jboss_adapter_installation.adoc:14]:
```suggestion
For more details on how to integrate {project_name} with JakartaEE applications running on latest Wildfly/EAP, take a look at the Jakarta EE quickstart in the {quickstartRepo_link}[Keycloak Quickstart GitHub Repository].
```

File: docs/documentation/securing_apps/topics/saml/java/jboss-adapter/jboss_adapter_installation.adoc [hunk 4]
```diff
@@ -1,130 +1,16 @@
...
-    </security-domains>
-----
-
-The security context is propagated to the EJB tier automatically.
-
-===== JBoss SSO
-
-{appserver_name} has built-in support for single sign-on for web applications deployed to the same {appserver_name}
-instance. This should not be enabled when using {project_name}.
+Below are the details about how to configure the SAML adapter secured by Galleon.
```
Comment (stianst) [inline docs/documentation/securing_apps/topics/saml/java/jboss-adapter/jboss_adapter_installation.adoc:16]:
```suggestion
Below are the details on how to configure the SAML adapter secured by Galleon.
```

File: docs/documentation/release_notes/topics/23_0_0.adoc (hunk 1 above)
Comment (mposolda) [reply docs/documentation/release_notes/topics/23_0_0.adoc]: Updated

File: docs/documentation/release_notes/topics/23_0_0.adoc [hunk 5]
```diff
@@ -1,3 +1,9 @@
+= OIDC and SAML adapter changes for WildFly and JBoss EAP
+
+OIDC adapter for the WildFly and JBoss EAP, which was deprecated in previous versions, is removed in this release. It is being replaced by the Elytron OIDC adapter,
+which is provided as part of the WildFly. The SAML adapter ZIP download for WildFly/EAP is removed as it is being replaced by the Galleon feature pack.
```
Comment (stianst) [inline docs/documentation/release_notes/topics/23_0_0.adoc:4]:
```suggestion
OIDC adapter for the WildFly and JBoss EAP, which was deprecated in previous versions, is removed in this release. It is being replaced by the Elytron OIDC adapter,
which is included in WildFly. The SAML adapter ZIP download for WildFly/EAP is removed as it is being replaced by the Galleon feature pack.
```

**Review by stianst — APPROVED** [review]

**Pull request merged**